docs/_tooling/extensions/header_service/header_service.py

```diff
@@ -0,0 +1,309 @@
+        "author": author,
+        "approver": data.get("Approved", "N/A"),
+        "reviewer": data.get("Reviewed", "N/A"),
+        "hash": hash,
+    }
+    return commit_data
+
+
+def _extract_github_data() -> dict[str, str | list[str]]:
+    """Retrieve data from GitHub and returns it as a dictionary.
+
+    :return: A dictionary containing the GitHub data.
+    """
+    empty_data: dict[str, str | list[str]] = {
+        "author": "N/A",
+        "approvers": "N/A",
+        "reviewers": "N/A",
+        "hash": "N/A",
+    }
+    try:
+        github: Github | None = None
+        auth = Auth.Token(_extract_github_token())
+        github = Github(auth=auth)
+        org: Organization.Organization = github.get_organization(_extract_org())
+        repo: Repository.Repository = github.get_repo(_extract_repo())
+        pr = repo.get_pull(int(_extract_pull_request()))
+        reviews = pr.get_reviews()
+        author = pr.user.login
+        data: dict[str, str | list[str]] = {
+            "author": author,
+            "approvers": _extract_approvers(reviews, _extract_team_info(org)),
+            "reviewers": _extract_reviewers(reviews, author),
+            "hash": pr.merge_commit_sha,
+        }
+    except Exception as ex:
+        print(f"Unable to extract Github information\n{ex}")
+        data = empty_data
+    finally:
+        if github:
+            github.close()
+    return data
+
+
+def _extract_org() -> str:
+    """Retrieve the organization from the GITHUB_REPOSITORY environment variable.
+
+    :return: The organization name.
+    """
+    return os.environ["GITHUB_REPOSITORY"].split("/")[0]
+
+
+def _extract_repo() -> str:
+    """Retrieve the repository from the GITHUB_REPOSITORY environment variable.
+
+    :return: The repository name.
+    """
+    return os.environ["GITHUB_REPOSITORY"]
+
+
+def _extract_github_token() -> str:
+    """Retrieve the GitHub token from the GH_TOKEN environment variable.
+
+    :return: The GitHub token.
+    """
+    return os.environ["GH_TOKEN"]
+
+
+def _extract_pull_request() -> str:
+    """Retrieve the pull request number from the GITHUB_REF_NAME environment variable.
+
+    :return: The pull request number.
+    """
+    return os.environ["GITHUB_REF_NAME"].split("/")[0]
+
+
+def _extract_approvers(
+    reviews: PaginatedList.PaginatedList[PullRequestReview.PullRequestReview],
+    team_info: dict[str, list[str]],
+) -> str:
+    """Retrieve a list of all approvers of the pull request.
+
+    :param reviews: The list of reviews.
+    :param team_info: The team information dictionary.
+    :return: A string containing the approvers.
+    """
+    approvers_set: set[str] = {
+        review.user.login for review in reviews if review.state == "APPROVED"
+    }
+    approvers_list: list[str] = _append_approver_teams(list(approvers_set), team_info)
+    return "; ".join(sorted(approvers_list))
+
+
+def _extract_team_info(org: Organization.Organization) -> dict[str, list[str]]:
```

Comment:
Names for functions which actively access the github api should clearly indicate that. However "extract" is also used to simple environment access.